core/new-gui/src/app/workspace/component/operator-panel/operator-label/operator-label.component.ts

```diff
@@ -16,21 +17,51 @@ import { OperatorSchema } from '../../../types/operator-schema.interface';
 })
 export class OperatorLabelComponent implements AfterViewInit {
 
+  @ViewChild('t') t: any; // t is the specific tooltip window for an operator
   @Input() operator?: OperatorSchema;
   public operatorLabelID: string;
+  private timer: any; // needed to add a delay to tooltip
+  private isHovering: boolean;
 
   constructor(
     private dragDropService: DragDropService
   ) {
     // generate a random ID for this DOM element
     this.operatorLabelID = 'texera-operator-label-' + uuid();
+
+    this.isHovering = false;
   }
 
   ngAfterViewInit() {
     if (! this.operator) {
       throw new Error('operator label component: operator is not specified');
     }
     this.dragDropService.registerOperatorLabelDrag(this.operatorLabelID, this.operator.operatorType);
+
+  }
+
+  // show the tooltip window after 1500ms
+  // var t means tooltip
+  public displayDescription(t: any): void {
+    this.isHovering = true;
+    const secondsCounter = interval(1500);
```

Comment:
1500 interval is too long

Reply:
Instead of using `rxjs interval`, I believe `rxjs timer` will be a better candidate to call here. Since according to the rxjs documentation, interval will trigger the event continuously. In this case, it will trigger this event can check whether `this.isHovering` is true every 1500 ms. As for `rxjs timer`, this will only trigger once, which might be enough for this scenario.

Here is the link to the `timer` API: https://www.learnrxjs.io/operators/creation/timer.html